### DISCUSSION TITLE
How to hide product viewer code?

### DISCUSSION THREAD

**Discussion by phovorka** [google/model-viewer#3244] [General]
Is any possibility how to hide my product customizer code from copying?

**Comment by elalish** [comment]
Generally the best way to hide JS code is to use a bundling/minifying tool that obfuscates. WASM is also pretty inscrutable. You might be better off with the legal system though; you own the copyright, and it's pretty easy to tell when someone has ripped off your JS.

**Reply by phovorka** [reply]
Thanks ... another question regarding this is how to hide GLB model source path?- for example this guys made this invisible...-> https://customizer.r2u.io/

**Reply by elalish** [reply]
I wouldn't call that hidden. It's just that the `src` was set in JS instead of HTML. The GLB can still be easily found and downloaded from the network tab in dev tools.

**Reply by clochardM33** [reply]
There isn't a free way to do it. As someone wrote "If it's in the browser it's with the client".
So to get out of that hole we went with a nasty file swap scheme where the src you actually want is swapped into a neutral file and then once model viewer has loaded it the neutral file is deleted. You have to do this in php if you want to keep it from the browser. Best of luck.
